File: gallery/src/main/java/com/qkopy/gallery/ui/common/BasePresenter.java
/*
 * Created by @ajithvgiri on 4/12/18 8:13 PM
 * Copyright (c) 2018 . All rights reserved.
 * Last modified 4/12/18 7:05 PM
 */

package com.qkopy.gallery.ui.common;

public class BasePresenter<T extends MvpView> {

    private T view;

    public void attachView(T view) {
        this.view = view;
    }

    public T getView() {
        return view;
    }

    public void detachView() {
        view = null;
    }

    protected boolean isViewAttached() {
        return view != null;
    }

}

File: gallery/src/main/java/com/qkopy/gallery/ui/common/BasePresenter.kt
package com.qkopy.gallery.ui.common

open class BasePresenter<T : MvpView?> {
    var view: T? = null
        private set

    fun attachView(view: T) {
        this.view = view
    }

    fun detachView() {
        view = null
    }

    protected val isViewAttached: Boolean
        protected get() = view != null
}